src/System.Private.CoreLib/shared/System/Environment.Win32.cs

```diff
@@ -194,7 +194,7 @@ public static string UserDomainName
                 // Domain names aren't typically long.
                 // https://support.microsoft.com/en-us/help/909264/naming-conventions-in-active-directory-for-computers-domains-sites-and
                 Span<char> initialDomainNameBuffer = stackalloc char[64];
```

Comment:
`builder` is not disposed above. If it happened to allocate a pooled array, it will leak.

Reply:
Same problem for in UserName property and domainBuilder.

Reply:
Fixed.